src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -143,3 +159,198 @@ const filterDecorators = (
+            member.initializer
+          )
+        )
+      );
+    } else {
+      // if it's not a class field that is decorated with a Stencil decorator then
+      // we just push it onto our class member list
+      updatedClassMembers.push(member);
+    }
+  }
+
+  // create or update a constructor which contains the initializing statements
+  // we created above
+  return updateConstructor(classNode, updatedClassMembers, statements);
+}
+
+/**
+ * Helper util for updating the constructor on a class declaration AST node.
+ *
+ * @param classNode the class node whose constructor we're going to be updated
+ * @param classMembers a list of class members for that class
+ * @param statements a list of statements which should be added to the
+ * constructor
+ * @returns a list of updated class elements
+ */
+export const updateConstructor = (
+  classNode: ts.ClassDeclaration,
+  classMembers: ts.ClassElement[],
+  statements: ts.Statement[]
+): ts.ClassElement[] => {
+  const constructorIndex = classMembers.findIndex((m) => m.kind === ts.SyntaxKind.Constructor);
+  const constructorMethod = classMembers[constructorIndex];
+
+  if (constructorIndex >= 0 && ts.isConstructorDeclaration(constructorMethod)) {
+    const hasSuper = constructorMethod.body.statements.some((s) => s.kind === ts.SyntaxKind.SuperKeyword);
+
+    if (!hasSuper && needsSuper(classNode)) {
+      statements = [createConstructorBodyWithSuper(), ...statements];
+    }
+
+    classMembers[constructorIndex] = ts.updateConstructor(
+      constructorMethod,
+      constructorMethod.decorators,
+      constructorMethod.modifiers,
+      constructorMethod.parameters,
+      ts.updateBlock(constructorMethod.body, statements)
+    );
+  } else {
+    // we don't seem to have a constructor, so let's create one and stick it
+    // into the array of class elements
+    if (needsSuper(classNode)) {
+      statements = [createConstructorBodyWithSuper(), ...statements];
+    }
+
+    classMembers = [
+      ts.createConstructor(undefined, undefined, undefined, ts.createBlock(statements, true)),
+      ...classMembers,
+    ];
+  }
+
+  return classMembers;
+};
+
+/**
+ * Check that a given class declaration should have a `super()` call in its
+ * constructor. This is something we can check by looking for a
+ * {@link ts.HeritageClause} on the class's AST node.
+ *
+ * @param classDeclaration a class declaration AST node
+ * @returns whether this class has parents or not
+ */
+const needsSuper = (classDeclaration: ts.ClassDeclaration): boolean =>
+  classDeclaration.heritageClauses && classDeclaration.heritageClauses.length > 0;
+
+/**
+ * Create a statement with a call to `super()` suitable for including in the body of a constructor.
+ * @returns a {@link ts.ExpressionStatement} node equivalent to `super()`
+ */
+const createConstructorBodyWithSuper = (): ts.ExpressionStatement => {
+  return ts.createExpressionStatement(ts.createCall(ts.createIdentifier('super'), undefined, undefined));
```

Comment:
Should we handle cases like https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-6.html#allowing-code-in-constructors-before-super?

Reply:
what would you suggest we do? right now this will basically add a `super()` call if one isn't present in the existing constructor or if we're creating a new one if the class inherits from another class. Right now it always gets added as the first statement in the constructor

Reply:
Hmmm - are you saying it's always added there today? If so, maybe I jumped the gun and we should punt on this for now.  If not, I wonder if we can find the index of the `super()` call in an existing ctor (if it exists) and use that? And if not we just put it at the start?